lib/fetch_book_data.rb

```diff
@@ -0,0 +1,53 @@
+require 'net/http'
+
+# Service object for pulling in book data from the CMS
+class FetchBookData
+  CMS_API_URL = Rails.application.secrets.cms_api_url
+  SUBJECTS_URL = "#{CMS_API_URL}snippets/subjects/?format=json"
+  TITLES_URL = "#{CMS_API_URL}v2/pages/?type=books.Book&format=json&limit=250&fields=title,book_subjects,book_state"
+  TIMEOUT = 1
+
+  def subjects
+    @subjects ||= fetch_subjects
```

Comment:
suggestion.   memoize the actual data from the call, but allow subjects and titles to be re-fetched, allow a flag that's defaulted to override this behavior to fetch live (this is just a consideration for multiple use of this object).  

what about caching the data (redis perhaps) ?  does each refresh of step4 need to fetch from the CMS for thousands of students a day?  how often do subjects and titles really change?   

have you you done performance testing on this call?  are you certain that thousands of educators at once signing up will be handled by this with quick response (hence the question about consideration of caching).  

something to think about.

Reply:
Right; I am going to cache the API response but for now just creating the FetchBookData service that calls the API. 

> memoize the actual data from the call, but allow subjects and titles to be re-fetched, allow a flag that's defaulted to override this behavior to fetch live (this is just a consideration for multiple use of this object).

Not entirely sure what you mean by this, or what the advantage is of "memoize the actual data from the call, but allow subjects and titles to be re-fetched" 🤔 .

Reply:
just making the suggestion to build in a little bit of flexibility into this class (optionally) to allow one to use it to fetch "live" or to fetch cached (perhaps by using a defaulted parameter).  this is not that big of a deal, esp since you're caching it anyway.

Reply:
I see. Yeah, for now I'm going to handle the caching outside of this object.